src/Trace/Trace.php

```diff
@@ -0,0 +1,211 @@
+     * @param array $spans [optional] Array of TraceSpan constructor arguments. See
+     *        {@see Google\Cloud\Trace\TraceSpan::__construct()} for configuration details.
+     * }
+     */
+    public function __construct(ConnectionInterface $connection, $projectId, $traceId = null, $spans = null)
+    {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->traceId = $traceId ?: $this->generateTraceId();
+        if ($spans) {
+            $this->spans = array_map(function ($span) {
+                return new TraceSpan($span);
+            }, $spans);
+        }
+    }
+
+    /**
+     * Retrieves the trace's id.
+     *
+     * @return string
+     */
+    public function traceId()
+    {
+        return $this->traceId;
+    }
+
+    /**
+     * Retrieves the trace's projectId.
+     *
+     * @return string
+     */
+    public function projectId()
+    {
+        return $this->projectId;
+    }
+
+    /**
+     * Set the trace's projectId
+     *
+     * @param string $projectId
+     */
+    public function setProjectId($projectId)
+    {
+        $this->projectId = $projectId;
+    }
+
+    /**
+     * Returns a serializable array representing this trace. If no span data
+     * is cached, a network request will be made to retrieve it.
+     *
+     * @return array
+     */
+    public function info()
+    {
+        if (!$this->spans) {
+            $this->reload();
+        }
+
+        return [
+            'projectId' => $this->projectId,
+            'traceId' => $this->traceId,
+            'spans' => array_map(function ($span) {
+                return $span->info();
+            }, $this->spans)
+        ];
+    }
+
+    /**
+     * Triggers a network request to load a span's details.
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces/get
+     */
+    public function reload()
+    {
+        $trace = $this->connection->getTrace([
+            'projectId' => $this->projectId,
+            'traceId' => $this->traceId
+        ]);
+
+        if (empty($trace)) {
+            throw new NotFoundException('Trace ID does not exist', 404);
+        }
+
+        $this->spans = array_map(function ($span) {
+            return new TraceSpan($span);
+        }, $trace['spans']);
+    }
+
+    /**
+     * Retrieves the spans for this trace.
+     *
+     * @return TraceSpan[]
+     */
+    public function spans()
+    {
+        return $this->spans;
+    }
+
+    /**
+     * Create an instance of {@see Google\Cloud\Trace\TraceSpan}
+     *
+     * @param array $options [optional] See {@see Google\Cloud\Trace\TraceSpan::__construct()}
+     *      for configuration details.
+     * @return TraceSpan
+     */
+    public function span(array $options = [])
+    {
+        return new TraceSpan($options);
+    }
+
+    /**
+     * Set the spans for this trace.
+     *
+     * @param TraceSpan[] $spans
+     */
+    public function setSpans(array $spans)
```

Comment:
Would it be beneficial to add an `addSpan(Span $span)` method as well, to push a single additional span onto an existing set?